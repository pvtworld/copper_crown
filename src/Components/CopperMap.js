import React, { Component } from 'react';
import {withGoogleMap, GoogleMap, Circle} from "react-google-maps";
import {geolocation} from '../Helpers/GeoHelpers';

const GameMap = withGoogleMap(props => (


    <GoogleMap
        center={props.center}
        defaultZoom={18}
        defaultCenter={props.center}
        onClick={props.onMapClick}
        options={{
            mapTypeControl: false,
            streetViewControl: false,
            // zoomControl: false,
        }}

    >
        <Circle
            center={props.center}
            radius={props.radius}
            options={{
                fillColor: `red`,
                fillOpacity: 0.90,
                strokeColor: `red`,
                strokeOpacity: 0.50,
                strokeWeight: 2,
            }}
        />

    </GoogleMap>
));


export default class CopperMap extends Component {
    constructor() {
        super();

        this.state = {
            center: {
                lat: 59.334591,
                lng: 18.063240,
            },
            content: null,
            radius: 5
        };
    }

    isUnmounted = false;

    handleMapClick = this.handleMapClick.bind(this);

    componentDidMount() {
        geolocation.getCurrentPosition((position) => {
            if (this.isUnmounted) {
                return;
            }
            this.setState({
                center: {
                    lat: position.coords.latitude,
                    lng: position.coords.longitude,
                },
                content: `Location found using HTML5.`,
            });

        }, (reason) => {
            if (this.isUnmounted) {
                return;
            }
            this.setState({
                center: {
                    lat: 59.334591,
                    lng: 18.063240,
                },
                content: `Error: The Geolocation service failed (${reason}).`,
            });
        });
    }

    componentWillUnmount() {
        this.isUnmounted = true;
    }

    handleMapClick(event) {
        // console.log("Latitude:" + event.latLng.lat());
        // console.log("Longitude:" + event.latLng.lng());
        console.log("Click outside circle not allowed");
    }

    render() {
        return (
            <div className="overlay">
                <div style={{height: `100%`}}>

                <GameMap
                    containerElement={
                        <div style={{ height: `100%` }} />
                    }
                    mapElement={
                        <div style={{ height: `100%` }} />
                    }
                    onMapClick={this.handleMapClick}
                    center={this.state.center}
                    content={this.state.content}
                    radius={this.state.radius}

                />

                </div>
            </div>
        );
    }
}