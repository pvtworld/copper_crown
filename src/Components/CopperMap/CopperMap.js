import React, {Component} from 'react';
import { geoError, geoOptions} from '../../Helpers/GeoHelpers';
import InfoContainer from "../InfoContainer/InfoContainer";
import GameMap from "../GameMap/GameMap";
import {connect} from 'react-redux';
import {firebaseConnect, dataToJS, pathToJS} from 'react-redux-firebase'
import { showUsernameModal } from '../../Redux/Actions/userActions';

class CopperMap extends Component {
    constructor() {
        super();
        this.state = {
            isMounted: false,
            center: {
                lat: 59.32571,
                lng: 18.07107,
            },
        };
    }

    geoTimer = null;

    geoLocationWatcher = this.geoLocationWatcher.bind(this);
    setMap = this.setMap.bind(this);
    onDragEnd = this.onDragEnd.bind(this);
    

    geoLocationWatcher() {
        if (navigator.geolocation) {

            let watchPositionId;

            const geoSucess = (position) => {
                if(this.state.isMounted){
                    this.map ? this.map.panTo({lat: position.coords.latitude, lng: position.coords.longitude}) : console.log("No map");
                    //this.setState({center: {lat: position.coords.latitude, lng: position.coords.longitude}})
                }

                navigator.geolocation.clearWatch(watchPositionId);
            }

            navigator.geolocation.getCurrentPosition(geoSucess, geoError, geoOptions);

            this.geoTimer = setInterval(function () {
                watchPositionId = navigator.geolocation.watchPosition(geoSucess, geoError, geoOptions);
            }, 5000);

        } else {
            alert("Turn on GPS")
        }
    }

    componentDidMount() {
        this.setState({
            isMounted: true
        })
        this.geoLocationWatcher();

    }

    componentWillUnmount() {
        this.setState({
            isMounted: false
        })
        clearInterval(this.geoTimer);
    }

    componentWillUpdate(){
        if(this.props.userInfo && (this.props.userInfo.username === 'NOT_SET') && (this.props.showNewUserModal !== true)){
            this.props.dispatch(showUsernameModal());
        }
    }

    setMap(map){
        
        this.map = map
        
    }

    onDragEnd(){
        if(this.map){
            
            this.map.panTo(this.state.center)
        }
        else{
            console.log("No map")
        }
    }


    render() {
        return (
            <div>
                <div className="map">
                    <div style={{height: `100%`}}>

                        <GameMap
                            containerElement={
                                <div style={{height: `100%`}}/>
                            }
                            mapElement={
                                <div style={{height: `100%`}}/>
                            }
                            center={this.state.center}
                            content={this.state.content}
                            radius={this.state.radius}
                            mapCallBack={this.setMap}
                            onDragEnd={this.onDragEnd}
                        />

                    </div>


                </div>
                <InfoContainer/>
            </div>
        );
    }
}
const mapStateToProps = (state, {auth}) => {
    return {
        userInfo: auth ? dataToJS(state.firebase, `users/${auth.uid}`) : undefined,
        showNewUserModal: state.showNewUserModal.show
    }
}

const propsConnected = connect(mapStateToProps)(CopperMap)

const wrappedPlayerInfo = firebaseConnect(
    ({auth}) => ([auth ? `users/${auth.uid}` : '/']))(propsConnected);

const authConnected = connect(
    (state) => ({
        auth: pathToJS(state.firebase, 'auth')
    })
)(wrappedPlayerInfo)

export default authConnected
