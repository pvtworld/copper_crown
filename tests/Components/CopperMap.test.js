import React, {Component} from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CopperMap from '../../src/Components/CopperMap/CopperMap'

beforeAll(() => {
    
});
describe('Render test', ()=> {
    it('renders a GameMap', () => {
        const component = shallow(<CopperMap/>);
        component.setState({authed : true});
        component.setState({userInfo: {points : 0}});
        
        // expect(toJson(component)).toMatchSnapshot();
    })
});