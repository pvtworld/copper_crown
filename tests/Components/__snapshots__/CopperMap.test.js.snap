exports[`Render test renders a GameMap 1`] = `
<div>
  <div
    className="map">
    <div
      style={
        Object {
          "height": "100%",
        }
      }>
      <withGoogleMap(Component)
        center={
          Object {
            "lat": 59.334591,
            "lng": 18.06324,
          }
        }
        containerElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            } />
        }
        content={null}
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            } />
        }
        onMapClick={[Function]}
        radius={5} />
    </div>
  </div>
  <UserScore />
</div>
`;
